packages/core/src/codewhisperer/ui/statusBar.ts

```diff
@@ -0,0 +1,108 @@
+import { createExitButton } from '../../shared/ui/buttons'
+import { isWeb } from '../../common/webUtils'
+
+function getAmazonQCodeWhispererNodes() {
+    const autoTriggerEnabled = CodeSuggestionsState.instance.isSuggestionsEnabled()
+    void vscode.commands.executeCommand('setContext', 'gumby.isTransformAvailable', false)
+    if (AuthUtil.instance.isConnectionExpired()) {
+        return [createReconnect('item'), createLearnMore()]
+    }
+
+    if (!AuthUtil.instance.isConnected()) {
+        return [createSignIn('item'), createLearnMore()]
+    }
+
+    if (vsCodeState.isFreeTierLimitReached) {
+        if (hasVendedIamCredentials()) {
+            return [createFreeTierLimitMet('item'), createOpenReferenceLog()]
+        }
+        return [createFreeTierLimitMet('item'), createOpenReferenceLog(), createSecurityScan()]
+    }
+
+    if (hasVendedIamCredentials()) {
+        return [createAutoSuggestions(autoTriggerEnabled), createOpenReferenceLog()]
+    }
+
+    if (AuthUtil.instance.isValidEnterpriseSsoInUse()) {
+        void vscode.commands.executeCommand('setContext', 'gumby.isTransformAvailable', true)
+    }
+
+    // TODO: Remove when web is supported for amazonq
+    let amazonq
+    if (!isWeb()) {
+        amazonq = require('../../amazonq/explorer/amazonQChildrenNodes')
+    }
+
+    return [
+        // CodeWhisperer
+        createSeparator('Inline Suggestions'),
+        createAutoSuggestions(autoTriggerEnabled),
+        ...(AuthUtil.instance.isValidEnterpriseSsoInUse() && AuthUtil.instance.isCustomizationFeatureEnabled
+            ? [createSelectCustomization()]
+            : []),
+        createOpenReferenceLog(),
+        createGettingStarted(), // "Learn" node : opens Learn CodeWhisperer page
+
+        // Amazon Q + others
+        createSeparator('Other Features'),
+        ...(amazonq ? [amazonq.switchToAmazonQNode('item')] : []),
+        createSecurityScan(),
+        ...(AuthUtil.instance.isValidEnterpriseSsoInUse() && amazonq ? [amazonq.createTransformByQ()] : []),
+    ]
+}
+
+export function getQuickPickItems(): DataQuickPickItem<string>[] {
+    const children = [
+        ...getAmazonQCodeWhispererNodes(),
+
+        // Generic Nodes
+        createSeparator('Connect / Help'),
+        createFeedbackNode(),
+        createGitHubNode(),
+        createDocumentationNode(),
+
+        // Add settings and signout
+        createSeparator(),
+        createSettingsNode(),
+        ...(AuthUtil.instance.isConnected() && !hasVendedIamCredentials() ? [createSignout()] : []),
+    ]
+
+    return children
+}
+
+export const listCodeWhispererCommandsId = 'aws.codewhisperer.listCommands'
+export const listCodeWhispererCommands = Commands.declare({ id: listCodeWhispererCommandsId }, () => () => {
+    return createQuickPick(getQuickPickItems(), {
+        title: 'Amazon Q (Preview) + CodeWhisperer',
+        buttons: [createExitButton()],
+        ignoreFocusOut: false,
+    }).prompt()
+})
```

Comment:
Should we force a refresh before presenting the menu?

Reply:
no. since the quick pick items are built whenever the menu item is opened, no refreshes are required.